HeinOnline.js

```diff
@@ -134,17 +130,17 @@ function scrapePage(doc, url) {
 			if (pdfPageURL) {
 				pdfPageURL = docParams.base + pdfPageURL;
 				// Z.debug(pdfPageURL)
-				ZU.doGet(pdfPageURL, function (pdfPage) {
+				ZU.processDocuments(pdfPageURL, function (pdfDoc) {
 					// Call to pdfPageURL prepares PDF for download via META refresh URL
 					var pdfURL = null;
-					var m = pdfPage.match(/<META.*URL="([^"]+)/);
+					var m = pdfDoc.querySelector('meta[http-equiv="Refresh"]');
 					// Z.debug(pdfPage)
 					// Z.debug(m)
 					if (m) {
-						pdfURL = docParams.base + m[1];
+						pdfURL = docParams.base + m.getAttribute('url');
```

Comment:
This isn't right, is it? `URL` isn't an attribute — it's a parameter within the `CONTENT` attribute. That's why we [split it](https://github.com/zotero/zotero/blob/a5fe49b86669a59bd65a20219b86e8a5c7e8e511/chrome/content/zotero/xpcom/http.js#L800) in the client.

Reply:
(And trim single quotes if present)

Reply:
From the source code of the redirect page:

```html
<META HTTP-EQUIV="Refresh" CONTENT="0" URL="PDFsearchable?handle=hein.journals/alterlj18&collection=journals&section=22&id=&print=section&sectioncount=3&ext=.pdf&nocover=&display=0">
```

Not technically correct HTML, but the site uses it (and browsers seem to handle it fine).

Reply:
Are we sure browsers handle it? There's also a JS redirect on that page.

Reply:
I don't think we should code to a spec violation. I can tell their dev that it's wrong.

Reply:
I think they just changed that based on our translator looking for `"`, but I assume that dates from a previous version of their site that used single quotes for the `CONTENT` parameter and double-quotes for the `URL` parameter. But making `URL` an attribute certainly isn't correct.

Reply:
Ah, right, good point. The browser does follow it, but it just reloads the same page rather than redirecting anywhere. Sure, if they fix it then I'll update the translator to parse spec-compliant HTML.

Reply:
Just to avoid temporary breakage when they fix it, we should probably push a version that can handle either a correct URL parameter within CONTENT or the URL attribute. And we can remove the latter once they've fixed it.

Reply:
OK, did that. Let me know if everything looks good to you.

Reply:
Looks good